nixos/modules/services/databases/ferretdb.nix

```diff
@@ -0,0 +1,97 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ferretdb;
+in
+{
+
+  meta.maintainers = with lib.maintainers; [ julienmalka camillemndn ];
+
+  options = {
+    services.ferretdb = {
+      enable = mkEnableOption (mdDoc "FerretDB, a truly Open Source MongoDB alternative.");
+
+      package = mkOption {
+        type = types.package;
+        example = literalExpression "pkgs.ferretdb";
+        default = pkgs.ferretdb;
+        defaultText = "pkgs.ferretdb";
+        description = lib.mdDoc "FerretDB package to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "ferretdb";
+        description = lib.mdDoc "User account under which FerretDB runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "ferretdb";
+        description = lib.mdDoc "Group under which FerretDB runs.";
+      };
+
+      settings = lib.mkOption {
+        type =
+          lib.types.submodule { freeformType = with lib.types; attrsOf str; };
+        default = {
+          FERRETDB_POSTGRESQL_URL = "postgres://localhost/ferretdb?host=/run/postgresql";
```

Comment:
This being set inside the option default means it will be overridden (instead of merged) once any other settings are defined with normal priority:

I suggest setting `services.ferretdb.settings.FERRETDB_POSTGRESQL_URL = lib.mkDefault ...;` inside the config block